addon/ActivityStreams.js

```diff
@@ -293,6 +292,9 @@ ActivityStreams.prototype = {
       case "bookmarkRemoved":
         this.broadcast(am.actions.Response("RECEIVE_BOOKMARK_REMOVED", data));
         break;
+      case "manyLinksChanged":
+        this.broadcast({type: "CLEAR_HISTORY"});
```

Comment:
I vaguely remember there were a few other reason this could be triggered, like a sync. That's fine since it would also be a valid refresh case, but might be better to name this something more generic and add a comment in the feed (i.e. this gets triggered by .... etc.)